subxt/src/backend/unstable/follow_stream_unpin.rs

```diff
@@ -74,20 +74,26 @@ impl<Hash: BlockHash> Stream for FollowStreamUnpin<Hash> {
         let mut this = self.as_mut();
 
         loop {
-            // Poll the unpin tasks while they are completing. if we get back None, then
-            // no tasks in the list, and if pending, we'll be woken when we can poll again.
-            if let Poll::Ready(Some(())) = this.unpin_futs.poll_next_unpin(cx) {
-                continue;
+            // Poll any queued unpin tasks.
+            let unpin_futs_are_pending = match this.unpin_futs.poll_next_unpin(cx) {
+                Poll::Ready(Some(())) => continue,
+                Poll::Ready(None) => false,
+                Poll::Pending => true,
             };
 
             // Poll the inner stream for the next event.
             let Poll::Ready(ev) = this.inner.poll_next_unpin(cx) else {
                 return Poll::Pending;
             };
 
-            // No more progress to be made if inner stream done.
             let Some(ev) = ev else {
-                return Poll::Ready(None);
+                // if the stream is done, but `unpin_futs` are still pending, then
+                // return pending here so that they are still driven to completion.
+                // Else, return `Ready(None)` to signal nothing left to do.
+                return match unpin_futs_are_pending {
+                    true => Poll::Pending,
```

Comment:
Here we just return pending if there are still unpin futures to resolve, even if the main stream is done.

Shouldn't really make a difference because we know that if the main stream is done, it'll have stopped and no more things will be pinned, but it felt better to not cancel the unpin futs midway just in case.